pkg/reconciler/credentials/s3/s3_secret.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package s3
+
+import (
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	"k8s.io/api/core/v1"
+)
+
+const (
+	AWSAccessKeyId         = "AWS_ACCESS_KEY_ID"
+	AWSSecretAccessKey     = "AWS_SECRET_ACCESS_KEY"
+	AWSAccessKeyIdName     = "awsAccessKeyID"
+	AWSSecretAccessKeyName = "awsSecretAccessKey"
+	AWSEndpointUrl         = "AWS_ENDPOINT_URL"
+	AWSRegion              = "AWS_REGION"
+	S3Endpoint             = "S3_ENDPOINT"
+	S3UseHttps             = "S3_USE_HTTPS"
+	S3VerifySSL            = "S3_VERIFY_SSL"
+)
+
+func AttachSecretEnvsAndVolume(secret *v1.Secret, configuration *v1alpha1.Configuration) {
```

Comment:
I'm not crazy about this attach pattern. It also doesn't like we're doing anything with volumes. 

Do you think it would be cleaner to have the code be more stateless? We could pass in a secret and generate env vars. The caller would be responsible for merging in the env vars.